app/main/ipcHandlers.ts

```diff
@@ -71,35 +72,50 @@ async function httpsGet(
 }
 
 const redirectUri = 'ConcordiumRedirectToken';
+const codeUriKey = 'code_uri=';
 
 function createExternalView(
     browserView: BrowserView,
     window: BrowserWindow,
     location: string,
     rect: Rectangle
-) {
+): Promise<ViewResponse> {
     return new Promise((resolve) => {
         window.setBrowserView(browserView);
         browserView.setBounds(rect);
         browserView.webContents
             .loadURL(location)
-            .then(() =>
-                browserView.webContents.on(
-                    'did-navigate',
-                    async (_e, url, httpResponseCode, httpStatusText) => {
+            .then(() => {
+                // Ignoring ts as it otherwise blocks us from using a custom
+                // event name ('abort').
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+                browserView.webContents.once('abort', async () => {
+                    resolve({
+                        status: ViewResponseStatus.Aborted,
+                    });
+                });
+
+                return browserView.webContents.on(
```

Comment:
This listener is still left on the BrowserView. This is probably irrelevant, because a user will not create enough identities (in one session) to make it matter.

Reply:
I remove it within the removal cleanup now.